### PR TITLE
all: make use of JujuComandBase and ModelCommand in core to simplify creating of httpbakery.Client with cookie jar

### PR DESCRIPTION
**Pull request by mattyw**

### CODE REVIEW

File: cmd/commands/commands.go [hunk 1]
```diff
@@ -26,14 +27,25 @@ type commandRegister interface {
...
-	r.Register(listbudgets.NewListBudgetsCommand())
-	r.Register(listplans.NewListPlansCommand())
-	r.Register(setbudget.NewSetBudgetCommand())
-	r.Register(setplan.NewSetPlanCommand())
-	r.Register(showbudget.NewShowBudgetCommand())
-	r.Register(updateallocation.NewUpdateAllocationCommand())
-	r.Register(listagreements.NewListAgreementsCommand())
+	register := func(c cmd.Command) {
+		switch c := c.(type) {
+		case modelcmd.ModelCommand:
```
Comment (cmars) [inline cmd/commands/commands.go]: I did not know that you could match interfaces in a type switch -- nice!

**Comment by cmars** [comment]
LGTM. Really like the replacement of rcmd.HttpCommand with the command base from juju/juju, thanks for doing this.

**Comment by alesstimec** [comment]
LGTM

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Pull request merged**